### PR TITLE
Examples + docs future

### PR DESCRIPTION
**Pull request by pedromorgan**
Manual to the example page

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,44 @@
...
 
-Each directory contains a revel application.
+Each sub directory contains a standalone [Revel](http://revel.github.io/manual)  application, eg
+
+狂欢是好的，保存UTF8 。保持微笑，享受..抱怨..请让它好看;-)
+
 
 ```
 go get github.com/revel/samples
 revel run github.com/revel/samples/booking
```
Comment (notzippy) [inline README.md]: Shouldn't these say examples now ?

Comment (pedromorgan) [inline README.md]: yep.. merge will sort it

**Comment by shawncatz** [comment]
I'm going to go through and redline these changes, bear with me...

**Comment by pedromorgan** [comment]
@shawncatz If u got the source then just commit to patch..

File: chat/README.md [hunk 2]
```diff
@@ -0,0 +1,330 @@
+Chat App Demo
+=========================
+The `Chat` app demonstrates ([browse the source](https://github.com/revel/samples/tree/master/chat)):
+
+* Using channels to implement a chat room with a [publish-subscribe](http://en.wikipedia.org/wiki/Publish%E2%80%93subscribe_pattern) model.
+* Using both `Comet` and [Websockets](../manual/websockets.html)
+
+Here are the key contents of the app:
```
Comment (shawncatz) [inline chat/README.md]: same as other examples, "Project Structure"

File: chat/README.md [hunk 3]
```diff
@@ -0,0 +1,330 @@
...
+```
+	chat/app/
+		chatroom	       # Chat room routines
+			chatroom.go
+
+		controllers
+			app.go         # The welcome screen, allowing user to pick a technology
```
Comment (shawncatz) [inline chat/README.md]: allowing user to choose from supported technologies

File: twitter-oauth/README.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
+Twitter OAuth with Revel example
+==================================
+
+The `twitter-oauth` app uses the `mrjones/oauth` library to demonstrate:
+
+* How to do the `oauth` dance to authenticate your app to use a `Twitter` account.
+* Fetching mentions for that Twitter account.
+* Tweeting on behalf of that Twitter account.
+
+The core contents of the app:
```
Comment (shawncatz) [inline twitter-oauth/README.md]: same as before, "Project Structure"

File: twitter-oauth/README.md [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+	},
+)
+```
+
+An overview of the process:
+
+1. The app generates a "request token" and sends the user to Twitter.
+2. The user authorizes the app.
+3. Twitter redirects the user to the provided redirect url, including an
+   "verifier" in the parameters.
```
Comment (shawncatz) [inline twitter-oauth/README.md]: "Twitter redirects the user to the provided URL, including a `verifier` parameter".

File: twitter-oauth/README.md [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+4. The app constructs a request to Twitter using the "request token" and
+   the "verifier", to which Twitter returns the "access token".
```
Comment (shawncatz) [inline twitter-oauth/README.md]: "constructs a request to Twitter using `request token` and `verifier`, which returns an `access token`."

File: twitter-oauth/README.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+5. The app henceforth uses the access token to operate Twitter on the user's behalf.
```
Comment (shawncatz) [inline twitter-oauth/README.md]: Henceforth, the app uses `access token` to ...

File: upload/README.md [hunk 8]
```diff
@@ -0,0 +1,8 @@
+Upload Files
+=========================
+
+TODO # pedro to cosument this
```
Comment (shawncatz) [inline upload/README.md]: "cosument" = "document" ?

**Comment by pedromorgan** [comment]
Yep agree with all changes and comments.. Merge it ..

File: README.md [hunk 9]
```diff
@@ -1,13 +1,44 @@
 # Sample Revel Applications
 
-Each directory contains a revel application.
+Each sub directory contains a standalone [Revel](http://revel.github.io/manual)  application, eg
+
+狂欢是好的，保存UTF8 。保持微笑，享受..抱怨..请让它好看;-)
```
Comment (brendensoares) [inline README.md]: what does this say?

File: booking/README.md [hunk 10]
```diff
@@ -23,16 +25,17 @@ The Booking sample app demonstrates ([browse the source](https://github.com/reve
...
 
 ## sqlite Installation
 
-The booking app uses [go-sqlite3](https://github.com/mattn/go-sqlite3) database driver (which wraps the native C library). 
-
+- The booking app uses [go-sqlite3](https://github.com/mattn/go-sqlite3) database driver, **which wraps the native C library**.
+- This means that the native c code needs to be installed first
 
-### To install on OSX:
+### Install sqlite on OSX:
```
Comment (brendensoares) [inline booking/README.md]: what about pkg-config?

File: booking/README.md [hunk 11]
```diff
@@ -41,31 +44,32 @@ The booking app uses [go-sqlite3](https://github.com/mattn/go-sqlite3) database
 $ brew install pkgconfig sqlite3
 ~~~
 
-### To install on Ubuntu:
-
-	$ sudo apt-get install sqlite3 libsqlite3-dev
-
-Once you have SQLite installed, it will be possible to run the booking app:
+### Install sqlite on Ubuntu:
```
Comment (brendensoares) [inline booking/README.md]: what about the booking install step?

File: README.md [hunk 12]
```diff
@@ -1,13 +1,44 @@
...
+
 
 ```
 go get github.com/revel/samples
 revel run github.com/revel/samples/booking
 ```
 
-Visit http://revel.github.io for more info
+* [Booking](booking.html) 
+  - A database-driven hotel-booking application, including user management
```
Comment (brendensoares) [inline README.md]: "hotel-booking" should be "hotel booking"

File: README.md [hunk 13]
```diff
@@ -1,13 +1,44 @@
 # Sample Revel Applications
 
-Each directory contains a revel application.
+Each sub directory contains a standalone [Revel](http://revel.github.io/manual)  application, eg
```
Comment (brendensoares) [inline README.md]: "sub-directory"

Comment (brendensoares) [inline README.md]: Actually, why do we need "sub" at all? In the context of this repo, "each directory" is accurate.

File: README.md [hunk 14]
```diff
@@ -1,13 +1,44 @@
...
 go get github.com/revel/samples
 revel run github.com/revel/samples/booking
 ```
 
-Visit http://revel.github.io for more info
+* [Booking](booking.html) 
+  - A database-driven hotel-booking application, including user management
+  
+* [Chat](chat.html) 
+  - A chat room demonstrating active refresh, long-polling (comet), and [websocket](http://revel.github.io/manual/websockets.html) implementations.
```
Comment (brendensoares) [inline README.md]: "Example of active refresh, Comet-style long polling, and WebSocket implementations."

File: README.md [hunk 15]
```diff
@@ -1,13 +1,44 @@
...
+  
+* [Validation](validation.html) 
+  - A demonstration of input [validation](http://revel.github.io/manual/validation.html).
```
Comment (brendensoares) [inline README.md]: "Example of Revel-based input validation."

Comment (brendensoares) [inline README.md]: "Example of Revel-based input validation."

File: README.md [hunk 16]
```diff
@@ -1,13 +1,44 @@
...
+  
+* [Upload](upload.html) 
+  - Demonstrates single and multiple file uploads to a webserver via http.
```
Comment (brendensoares) [inline README.md]: "Example of single and multiple file uploading."

File: README.md [hunk 17]
```diff
@@ -1,13 +1,44 @@
...
+  
+* [Twitter OAuth](twitter-oauth.html) 
+  - Display `mentions` and allows `posting` to a Twitter account using OAuth.
```
Comment (brendensoares) [inline README.md]: "Example of integrating with the Twitter OAuth API."

File: README.md [hunk 18]
```diff
@@ -1,13 +1,44 @@
...
+  
+* [Facebook OAuth2](facebook-oauth2.html) 
+  - Display Facebook user information using OAuth2.
```
Comment (brendensoares) [inline README.md]: "Example of integrating with the FaceBook OAuth2 API."

File: README.md [hunk 19]
```diff
@@ -1,13 +1,44 @@
...
+
+
 
 ## Contribute
-If you got a good example application then please make a pull request ;-)
+- If you spot and idiot mistake etc, then please make some noise
+- If you have a good sample application idea, then make some noise
+- If you got an app, then please make a pull request 
+- ta ;-)))))))))))))))
+- Early to bed, early to rise, makes a person, healthy and wise .. (code workz)
```
Comment (brendensoares) [inline README.md]: Absolutely agree. We need to keep it professional. You can be goofy in chat, but this style of language is distracting from the purpose of the words. We're not trying to convey a friendly message, we're trying to teach. Keep it simple.

File: booking/README.md [hunk 20]
```diff
@@ -1,14 +1,16 @@
-Hotel Bookings 
+Hotel Booking Example
 ===============================
 
-The Booking sample app demonstrates ([browse the source](https://github.com/revel/samples/tree/master/booking)):
+The Hotel Booking sample app demonstrates ([browse the source](https://github.com/revel/samples/tree/master/booking)):
```
Comment (brendensoares) [inline booking/README.md]: "example app"

File: booking/README.md [hunk 21]
```diff
@@ -1,14 +1,16 @@
...
 
 * Using an SQL (SQLite) database and configuring the Revel DB module.
 * Using the third party [GORP](https://github.com/coopernurse/gorp) *ORM-ish* library
 * [Interceptors](../manual/interceptors.html) for checking that an user is logged in.
 * Using [validation](../manual/validation) and displaying inline errors
 
 
+Here's a quick summary of the files and purpose
```
Comment (brendensoares) [inline booking/README.md]: Agreed

File: booking/README.md [hunk 22]
```diff
@@ -23,16 +25,17 @@ The Booking sample app demonstrates ([browse the source](https://github.com/reve
 
 		views
 			...
+```
 
-
-
+# Database Install and Setup
+This example used [sqlite](https://www.sqlite.org/), but u can easily adopt code to your needs with mysql, postgres, et all.
```
Comment (brendensoares) [inline booking/README.md]: "et al" but "etc" is shorter. Also, "u"?? Further, why is this relevant? Isn't this implied?

File: booking/README.md [hunk 23]
```diff
@@ -23,16 +25,17 @@ The Booking sample app demonstrates ([browse the source](https://github.com/reve
...
 
 ## sqlite Installation
 
-The booking app uses [go-sqlite3](https://github.com/mattn/go-sqlite3) database driver (which wraps the native C library). 
-
+- The booking app uses [go-sqlite3](https://github.com/mattn/go-sqlite3) database driver, **which wraps the native C library**.
+- This means that the native c code needs to be installed first
```
Comment (brendensoares) [inline booking/README.md]: "C" is a proper pronoun

File: booking/README.md [hunk 24]
```diff
@@ -41,31 +44,32 @@ The booking app uses [go-sqlite3](https://github.com/mattn/go-sqlite3) database
...
 
+Once SQLite is installed, it will be possible to run the booking app:
+```sh
 	$ revel run github.com/revel/samples/booking
+```
 
 ## Database / Gorp Plugin
 
-[`app/controllers/gorp.go`](https://github.com/revel/samples/blob/master/booking/app/controllers/gorp.go) defines `GorpPlugin`, which is a plugin that does a couple things:
+[`app/controllers/gorp.go`](https://github.com/revel/samples/blob/master/booking/app/controllers/gorp.go) defines `GorpPlugin`, which is a plugin that does a few things:
```
Comment (brendensoares) [inline booking/README.md]: It doesn't really implement anything and it's also not a plugin. It's a controller with functionality that can be embedded into other controllers. Plugin is not a concept in Revel, so we shouldn't use it.

File: chat/README.md [hunk 25]
```diff
@@ -0,0 +1,330 @@
+Chat App Demo
+=========================
+The `Chat` app demonstrates ([browse the source](https://github.com/revel/samples/tree/master/chat)):
```
Comment (brendensoares) [inline chat/README.md]: I'm not liking the "browser the source" links. Is there a better place for them?

File: chat/README.md (hunk 3 above)
Comment (brendensoares) [inline chat/README.md]: "welcome screen"? Is there a better term for that?

File: chat/README.md [hunk 26]
```diff
@@ -0,0 +1,330 @@
...
+			chatroom.go
+
+		controllers
+			app.go         # The welcome screen, allowing user to pick a technology
+			refresh.go     # Handlers for the "Active Refresh" chat demo
+			longpolling.go # Handlers for the "Long polling" ("Comet") chat demo
+			websocket.go   # Handlers for the "Websocket" chat demo
+
+		views
+			                # HTML and Javascript
```
Comment (brendensoares) [inline chat/README.md]: is there JS in the views?

File: twitter-oauth/README.md [hunk 27]
```diff
@@ -0,0 +1,46 @@
+Twitter OAuth with Revel example
+==================================
+
+The `twitter-oauth` app uses the `mrjones/oauth` library to demonstrate:
+
+* How to do the `oauth` dance to authenticate your app to use a `Twitter` account.
```
Comment (brendensoares) [inline twitter-oauth/README.md]: "OAuth" is the correct capitalization. "dance" is not the correct term here.

File: upload/README.md (hunk 8 above)
Comment (brendensoares) [inline upload/README.md]: or consummate...

Comment (brendensoares) [inline upload/README.md]: We need to remove this TODO or do it, quickly.

**Pull request closed without merging**